src/util/system.cpp

```diff
@@ -773,6 +773,11 @@ const fs::path &GetDataDir(bool fNetSpecific)
     return path;
 }
 
+bool CheckDataDirOption()
+{
+    return gArgs.IsArgSet("-datadir") ? fs::is_directory(fs::system_complete(gArgs.GetArg("-datadir", ""))) : true;
```

Comment:
This doesn't check the default data dir.

Reply:
You are right. This behavior is international.

Reply:
In commit "Add CheckDataDirOption() function" (a45f692366d7bb4814af83e3b645c9e5a5a734e6)

I don't think `IsArgSet` is the right function to call here. If somebody has a `datadir` option specified in the config file, but wants to unset it on the command line by passing `-nodatadir` or `-datadir=""`, it seems like this should be allowed. Would suggest:

```
std::string datadir = gArgs.GetArg("-datadir", "");
return datadir.empty() || fs::is_directory(fs::system_complete(datadir));
```

And a corresponding change in GetDataDir()

Reply:
> I don't think IsArgSet is the right function to call here. If somebody has a datadir option specified in the config file, but wants to unset it on the command line by passing -nodatadir or -datadir="", it seems like this should be allowed.

Good catch! Agree.